app/scripts/metamask-controller.test.js

```diff
@@ -509,6 +511,31 @@ describe('MetaMaskController', function () {
     });
   });
 
+  describe('getPrimaryKeyringMnemonic', function () {
+    it('should return a mnemonic as a string', function () {
+      const mockMnemonic =
+        'above mercy benefit hospital call oval domain student sphere interest argue shock';
+      const mnemonicIndices = mockMnemonic
+        .split(' ')
+        .map((word) => englishWordlist.indexOf(word));
+      const uint8ArrayMnemonic = new Uint8Array(
+        new Uint16Array(mnemonicIndices).buffer,
+      );
+
+      const mockHDKeyring = {
+        type: 'HD Key Tree',
+        mnemonic: uint8ArrayMnemonic,
+      };
+      sinon
+        .stub(metamaskController.keyringController, 'getKeyringsByType')
+        .returns([mockHDKeyring]);
```

Comment:
This mocks the way `mnemonics` are stored on HD keyrings - as `Uint8Arrays` of the indices of the words relative to the wordlist -  as of https://github.com/MetaMask/eth-hd-keyring/pull/67. This test then demonstrates that even though the mnemonic is stored in this way, the `getPrimaryKeyringMnemonic` method on `MetamaskController` returns the mnemonic in it's string form, since this is required by Snaps for now.